src/services/services.ts

```diff
@@ -4316,6 +4361,159 @@ namespace ts {
                     }
                 }
             }
+
+            function getStringLiteralCompletionEntriesFromModuleNames(node: StringLiteral) {
+                const literalValue = node.text;
+                let result: CompletionEntry[];
+
+                const isRelativePath = startsWith(literalValue, ".");
+                const scriptDir = getDirectoryPath(node.getSourceFile().path);
+                if (isRelativePath || isRootedDiskPath(literalValue)) {
+                    result = getCompletionEntriesForDirectoryFragment(literalValue, scriptDir, getSupportedExtensions(program.getCompilerOptions()), /*includeExtensions*/false);
+                }
+                else {
+                    // Check for node modules
+                    result = getCompletionEntriesForNonRelativeModules(literalValue, scriptDir);
+                }
+
+                return {
+                    isMemberCompletion: false,
+                    isNewIdentifierLocation: false,
+                    entries: result
+                };
+            }
+
+            function getCompletionEntriesForDirectoryFragment(fragment: string, scriptPath: string, extensions: string[], includeExtensions: boolean): CompletionEntry[] {
+                // Complete the path by looking for source files and directories
+                const result: CompletionEntry[] = [];
+
+                const toComplete = getBaseFileName(fragment);
+                const absolutePath = normalizeSlashes(host.resolvePath(isRootedDiskPath(fragment) ? fragment : combinePaths(scriptPath, fragment)));
+                const baseDir = toComplete ? getDirectoryPath(absolutePath) : absolutePath;
+
+
+                if (directoryProbablyExists(baseDir, host)) {
+                    // Enumerate the available files
+                    const files = host.readDirectory(baseDir, extensions, /*exclude*/undefined, /*include*/["./*"]);
+                    forEach(files, (f) => {
+                        const fName = includeExtensions ? getBaseFileName(f) : removeFileExtension(getBaseFileName(f));
+
+                        if (startsWith(fName, toComplete)) {

```

Comment:
this means in VS you need to get completions (ctrl+space) after a backspace if the completion list is visible?